include/gridtools/gcl/GCL.hpp

```diff
@@ -31,7 +31,7 @@ namespace gridtools {
 
         inline int procs() { return impl_::procs_holder(); }
 
-        inline void init(int argc, char **argv) {
+        inline void init(int argc = 1, char **argv = 0) {
```

Comment:
```suggestion
        inline void init(int argc = 0, char **argv = nullptr) {
```
`argc` has to be zero I guess as `argv[0]` is invalid.

Reply:
Sounds reasonable, but this was the default in GT before the refactoring. So not sure if it has side-effects. :(

Reply:
But passing `argc = 1` with a `argv` being a nullpointer to `MPI_Init(&argc, &argv)` clearly looks like a bug to me. The chance that MPI ever wants to touch `argv[0]` is maybe small, but still… By the way: starting from MPI2, calling `MPI_Init()` without arguments must also be supported, so maybe we could also do this.

Reply:
@mbianco Ok, if we drop the arguments completely in the `gcl::init()`? I.e. always use the `MPI_Init(nullptr, nullptr)` or do you see a problem?

Reply:
Just came across this. How does `gcl::init`  behaves in multithreaded environment? I wonder whether it should be gcl which initializes MPI, because it doesn't know how it should do MPI initialization. Basically, now we expect that a user of GCL initializes MPI himself if he uses any threads (MPI_Init = MPI_Init_thread with MPI_THREAD_SINGLE), otherwise he relies on the MPI library working for multithreading, too.

Reply:
Probably the problem is that originally gcl was intended for more than just MPI so wrapping setup was a useful design decision. Now, I guess some decisions don't make sense anymore. The future is ghex. ;)

Reply:
Sounds reasonable to me. We provide a shortcut for the default case (MPI_Init) and if the user ever wants to do more fancy stuff, MPI_Init_thread can be called manually, i.e., we do not restrict any use cases. As GCL will be replaced by GHEX in the longer term, I would not bother about adding additional init overloads now…

Reply:
ahhh... this is why I came across this. this is gcl, not ghex :)